### PR TITLE
Added flatten method to the Paddle frontend

### PR DESCRIPTION
**Pull request by he11owthere**
closes https://github.com/unifyai/ivy/issues/22582

image

### CODE REVIEW

**Comment by he11owthere** [comment]
Please remove "bfloat16" from `with_supported_dtypes` decorator before running the tests until the issue with `ivy.Array.repr` method has been fixed as mentioned by Ved in https://github.com/unifyai/ivy/pull/21447. Thank you! 🙂

**Comment by aparajith21** [comment]
Hi, could you please comment `add_frontend_checklist`, go over the items there, and check if they're all good and go over any test failures locally. Please request review after that and I will check or let me know if there's any help needed meanwhile!

**Comment by he11owthere** [comment]
add_frontend_checklist

**Comment by aparajith21** [comment]
Hey, seems there are some failures on the CI. Could you please check these locally and see what shows up there. Further, it would be great if you could check off the items in the front end checklist and confirm everything has been completed properly before requesting review. Same applies for the other PR! Thanks for taking the time and contributing!

**Comment by he11owthere** [comment]
Hi @aparajith21, all tests are passing locally, and I've thoroughly reviewed everything. This is my first time encountering this process after contributing for a while. I've tried to check for any inconsistencies and test checks, cannot seem to find what's wrong. Could you please mention if there's a specific problem to this PR, I'll try to fix it. 

image

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 1]
```diff
@@ -96,6 +96,28 @@ def reshape(self, *args, shape=None):
     def dim(self):
         return self.ivy_array.ndim
 
+    @with_supported_dtypes(
```
Comment (aparajith21) [inline ivy/functional/frontends/paddle/tensor/tensor.py:99]: The issue may be related to this, and you may want to change this to `unsupported` and check if tests pass on the CI

**Pull request merged**